Floofjaran

Cprn-Sprkl opened the pull request:
# Description

A port of Tajara ##1647 (https://github.com/Simple-Station/Einstein-Engines/pull/1647) for Floofstation.
Adds Tajara and two of their languages. Increases their customization options to be in line with other Floof species including more markings and wider fur colour range.

The traits for the Tajaran languages have been modified to be in line with other language traits in terms of points, and have a wider pool of acceptable species that can take them.

---

# Changelog

:cl: Sprkl
- add: The Tajara have arrived!

SX-7 committed:
Tajara (#1647)

~~On user end Felinids are removed and replaced with Tajara, mostly
inspired by the ParadiseStation sprites.~~

~~Most things from Felinids are ported over, either being untouched (in
which case they're left in the nyanotrasen directory), or somewhat
modified and copied to tajara directories. This is done because both
shadowkin and humans reference these files quite a bit, so we don't want
to break anything forwards/backwards,~~

~~Under the hood, felinid components are untouched, just simply set to
`abstract: true` and `roundstart: false`, once again to prevent any
breakage.~~

There's been like 50 changes all the ways, but basically, felinids are
staying, tajara get a few languages, they also are more "specialized"
than felinids due to big upsides and downsides (funny :)

---

- [X] Add tajara
~~- [X] Hide felinids~~
~~- [x] Analyze and find now-orphaned felinid
protorypes/textures/references and remove them (optional)~~
- [x] Fix graphical bugs (if any)
- [x] Ensure the code structure is compliant with repo practices

---

<details><summary><h1>Media</h1></summary>
<p>

In captain attire

![{3169C46E-BABB-466F-BD75-D8A00D8F9105}](https://github.com/user-attachments/assets/74a7c43f-9d5d-440f-9c18-f1eb386ad3d7)
Testing emotes and languages

![{7EE18197-E788-46BF-9DA3-97B0718A2F0B}](https://github.com/user-attachments/assets/74dd9c17-92a2-4760-99f8-1bb893fa969d)
Emotes still working, same with hairballs

![{346EB863-EF92-4D82-8E72-409437372DC3}](https://github.com/user-attachments/assets/91b61aae-d460-4373-a0ba-79cbd64f8ec4)
Few character selector pictures
- The previously shown off tajaran

![{559C39A8-5333-4787-A3B7-2F882EB6B5E2}](https://github.com/user-attachments/assets/38d77c48-caa2-461f-858a-a6b1a235cc0f)
- Markingless male body

![{9D762061-C1F8-40A5-9F14-5809E803820C}](https://github.com/user-attachments/assets/719c18ba-0230-4b17-9166-fdd15cb67526)

![{98AFAEF1-D49A-4951-A18E-5E55ACC05A26}](https://github.com/user-attachments/assets/6f848c0a-9a1e-4aed-aeb7-0bbf7c1dab62)
- Markingless female body

![{8F42252C-DDC4-43FD-8B8F-BFF0FBD971CD}](https://github.com/user-attachments/assets/92ea11bc-2ddc-4569-b166-0380b34ee946)

![{EE2B152D-97AD-4C39-B51E-8D89E4C4B097}](https://github.com/user-attachments/assets/c6f3a8ee-20be-486a-bea8-b862cf2ea12e)
- Traits are updated, as is the case with other texts

![{6F1E0309-7262-48BA-ABF6-C2048E087437}](https://github.com/user-attachments/assets/778ede7d-3069-4b81-b09d-32aadb6c6b76)
~~- And a neon colored specimen to show off some markings~~ No longer
the case, fur hue is clamped to some reasonable colors

![{381F7D38-1ABE-4434-8F29-95908E8F1A4B}](https://github.com/user-attachments/assets/1ef52004-db37-4127-a113-0b5640087cc1)

</p>
</details>

---

:cl:
- add: Added Tajara and related content

---------

Signed-off-by: SX-7 <92227810+SX-7@users.noreply.github.com>
Signed-off-by: VMSolidus <evilexecutive@gmail.com>
Co-authored-by: sleepyyapril <123355664+sleepyyapril@users.noreply.github.com>
Co-authored-by: VMSolidus <evilexecutive@gmail.com>

Cprn-Sprkl committed:
Give Tajara More Marks

Let's tajara use the same markings as felinids.

Cprn-Sprkl committed:
Fix duplicate trait

Cprn-Sprkl committed:
Bring language traits in line

Cprn-Sprkl committed:
fix chat language test fail

Cprn-Sprkl committed:
Tajara Floof changes

Widened fur colour options, disabled three of the five new languages, widened the list of species that could learn the remaining languages, removed extremely specific bits of lore from language descriptions allowing them to a be a wider base for players with other ideas, removed translator for now disabled languages.

Cprn-Sprkl committed:
Merge branch 'master' into floofjaran

Memeji reviewed (approved):
Tests pass, think it should be good.
Will need the other's input.

M3739 commented on Resources/Prototypes/_EE/Entities/Mobs/Species/tajaran.yml:
```diff
@@ -0,0 +1,129 @@
+    spawned:
+    - id: FoodMeatHuman
+      amount: 5
+  - type: TemperatureProtection # fur is a good insulator, actually
+    coefficient: 0.1
```
Restrict them from taking the `TemperatureTolerance` perk as they already have it innately.

M3739 reviewed (changes requested):
A few things to address.

Interestingly enough, it would seem that the languages have uh... a bit of a nuance to them: 
<img width="922" height="408" alt="image" src="https://github.com/user-attachments/assets/f1a62cae-929c-407b-8eb7-940b87fa1c5e" />

It seems like it is outputting both the verb and the actual sentence itself, instead of showing only the verb for non-understanding observers, and only the actual message for those that understand it.

Other than one other issue, it seems to check out on my end.

Cprn-Sprkl commented:
The original PR had five new Tajaran languages, which was deemed too many for one species. I cut it down to the two most interesting ones and opened them up to any species with both ears and tails.

One is a pure sign language, the other is a combination of body language and vocalizations. That's why it outputs a bit differently than other languages.

<img width="443" height="72" alt="image" src="https://github.com/user-attachments/assets/f0efeb70-7f7a-4abd-b707-f52d28b56dea" />
<img width="464" height="44" alt="image" src="https://github.com/user-attachments/assets/f4f945d5-30d6-4603-a823-cca4a82ccdb0" />

Cprn-Sprkl committed:
Restrict trait

Restricts Tajar from taking temperature tolerance due to already having temperature protection.

M3739 reviewed (approved):
I see. In that regard, I'll take it as is. I just had to quickly verify that it was being handled by radio communications properly. A little bit cursed as it only transmits the vocal part, but it works.

Cprn-Sprkl committed:
Merge branch 'master' into floofjaran
